### PR TITLE
Add 2023 DC tax rules

### LINKED ISSUE #4171: Add 2023 DC Tax Rules
2023 Income Tax Form Instructions: https://otr.cfo.dc.gov/sites/default/files/dc/sites/otr/publication/attachments/2023_D40_Book_Final_012324.pdf

### PR DESCRIPTION
**Pull request by shuyu7zhan**
Fixes #4171

### CODE REVIEW

File: policyengine_us/parameters/gov/states/dc/tax/income/credits/eitc/without_children/phase_out/start.yaml [hunk 1]
```diff
@@ -10,6 +10,9 @@ metadata:
       href: https://otr.cfo.dc.gov/sites/default/files/dc/sites/otr/publication/attachments/52926_D-40_12.21.21_Final_Rev011122.pdf#page=24
     - title: 2022 DC Form D-40 Booklet, Page 23, Line 5
       href: https://otr.cfo.dc.gov/sites/default/files/dc/sites/otr/publication/attachments/2022_D-40_Booklet_Final_blk_01_23_23_Ordc.pdf#page=23
+    - title: 2023 DC Form D-40 Booklet, Page 23, Line 5
+      href: https://otr.cfo.dc.gov/sites/default/files/dc/sites/otr/publication/attachments/2023_D40_Book_Final_012324.pdf#page=23
 values:
   2021-01-01: 19_743
   2022-01-01: 20_532
+  2022-01-01: 21_888
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/dc/tax/income/credits/eitc/without_children/phase_out/start.yaml:18]: 2022 twice

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: Does this additional standard deduction exist?
[Form](https://otr.cfo.dc.gov/sites/default/files/dc/sites/otr/publication/attachments/2023_D40_Book_Final_012324.pdf#page=34)

image

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: changelog_entry.yaml
Comment (PavelMakarchuk) [reply changelog_entry.yaml:1]: Just checked, it is the federal standard deduction, nothing needs to change

**Comment by PavelMakarchuk** [comment]
@shuyu7zhan could you summarize the changes in the issue #4171 - just a quick comment about what changed and what to

File: policyengine_us/parameters/gov/states/dc/tax/income/credits/ptc/fraction_elderly.yaml [hunk 2]
```diff
@@ -9,6 +9,8 @@ metadata:
       href: https://otr.cfo.dc.gov/sites/default/files/dc/sites/otr/publication/attachments/52926_D-40_12.21.21_Final_Rev011122.pdf#page=49
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/dc/tax/income/credits/ptc/fraction_elderly.yaml:9]: could you add to the metadata:
uprating: gov.irs.uprating

This allows us to predict the values for future dates

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/dc/tax/income/credits/ptc/fraction_elderly.yaml:9]: Could you do this for all yaml files that seem to increase every year with inflation

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/dc/tax/income/credits/ptc/fraction_elderly.yaml (hunk 2 above)
Comment (MaxGhenis) [reply policyengine_us/parameters/gov/states/dc/tax/income/credits/ptc/fraction_elderly.yaml:9]: note https://github.com/PolicyEngine/policyengine-core/issues/163 though

**Comment by shuyu7zhan** [comment]
Sure will do today… been busy with work for the past two days.

**Comment by shuyu7zhan** [comment]
Summary: 2023 DC tax rules updates
1. Credits - EITC - phase-out threshold for filers without qualifying children: $21888
2. Credits - KCCATC - taxable income limit: $167400 ($83700 for separate filers)
3. Credits - KCCATC - maximum amount per eligible child: $1115
4. Credits - PTC - fraction of US AGI for elderly: $83700
5. Credits - PTC - fraction of US AGI for non-elderly: $61300
6. Credits - PTC - maximum amount: $1325

File: policyengine_us/parameters/gov/states/dc/tax/income/credits/ptc/fraction_elderly.yaml (hunk 2 above)
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/dc/tax/income/credits/ptc/fraction_elderly.yaml:9]: yes, @shuyu7zhan could you add the uprating to the thresholds in the yaml files if they are updated with inflation 
See [CA](https://github.com/PolicyEngine/policyengine-us/pull/3770) as an example

File: policyengine_us/parameters/gov/states/dc/tax/income/credits/ptc/fraction_nonelderly.yaml [hunk 3]
```diff
@@ -4,6 +4,7 @@ metadata:
   threshold_period: year
   threshold_unit: currency-USD  # federal AGI
   amount_unit: /1
+  uprating: gov.irs.uprating
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/dc/tax/income/credits/ptc/fraction_nonelderly.yaml:7]: this won't currently work - it'll uprate both threshold and amount pending a core fix. see CA as an example for the interim

Comment (shuyu7zhan) [reply policyengine_us/parameters/gov/states/dc/tax/income/credits/ptc/fraction_nonelderly.yaml]: got u, just changed the two threshold-related uprating

File: policyengine_us/parameters/gov/states/dc/tax/income/credits/ptc/fraction_elderly.yaml [hunk 4]
```diff
@@ -17,5 +19,11 @@ brackets:
   - threshold:
       2021-01-01: 76_700
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/dc/tax/income/credits/ptc/fraction_elderly.yaml:20]: also do this elsewhere. if we have the `values:` all the values need to be nested under that

(@nikhilwoodruff i'm surprised this didn't break things..)
```suggestion
      values:
        2021-01-01: 76_700
        2022-01-01: 78_600
        2023-01-01: 83_700
      metadata:
        uprating: gov.irs.uprating
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/dc/tax/income/credits/ptc/max.yaml [hunk 5]
```diff
@@ -3,11 +3,21 @@ metadata:
   unit: currency-USD
   label: DC property tax credit maximum amount
   period: year
+  uprating:
+    parameter: gov.irs.uprating
+    rounding:
+      type: downwards
+      interval: 5
   reference:
+    - title: Code of the District of Columbia § 47–1806.06
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/dc/tax/income/credits/ptc/max.yaml:12]: could you add the subsection?

File: policyengine_us/parameters/gov/states/dc/tax/income/credits/ptc/fraction_nonelderly.yaml [hunk 6]
```diff
@@ -5,10 +5,14 @@ metadata:
   threshold_unit: currency-USD  # federal AGI
   amount_unit: /1
   reference:
+    - title: Code of the District of Columbia § 47–1806.06
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/dc/tax/income/credits/ptc/fraction_nonelderly.yaml:8]: could you add a subsection?

File: policyengine_us/parameters/gov/states/dc/tax/income/credits/ptc/max.yaml (hunk 5 above)
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/dc/tax/income/credits/ptc/max.yaml:12]: all legal code files

File: policyengine_us/parameters/gov/states/dc/tax/income/credits/refundable.yaml [hunk 7]
```diff
@@ -7,6 +7,8 @@ metadata:
       href: https://otr.cfo.dc.gov/sites/default/files/dc/sites/otr/publication/attachments/52926_D-40_12.21.21_Final_Rev011122.pdf#page=36
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/dc/tax/income/credits/refundable.yaml:7]: Can you change the unit to "`list`" and add a `period: year`

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/dc/tax/income/credits/ptc/fraction_elderly.yaml [hunk 8]
```diff
@@ -5,17 +5,25 @@ metadata:
...
       2021-01-01: 0.03
   - threshold:
-      2021-01-01: 76_700
-      2022-01-01: 78_600
+      values:
+        2021-01-01: 76_700
+        2022-01-01: 78_600
+        2023-01-01: 83_700
+      metadata:
+        uprating: gov.irs.uprating
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/dc/tax/income/credits/ptc/fraction_elderly.yaml:27]: check if we need rounding here, based on legal code

File: policyengine_us/parameters/gov/states/dc/tax/income/credits/ptc/fraction_nonelderly.yaml [hunk 9]
```diff
@@ -23,7 +27,11 @@ brackets:
     amount:
       2021-01-01: 0.05
   - threshold:
-      2021-01-01: 56_200
-      2022-01-01: 57_600
+      values:
+        2021-01-01: 56_200
+        2022-01-01: 57_600
+        2023-01-01: 61_300
+      metadata:
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/dc/tax/income/credits/ptc/fraction_nonelderly.yaml:34]: need to add rounding:
image

File: policyengine_us/parameters/gov/states/dc/tax/income/credits/ptc/fraction_elderly.yaml (hunk 8 above)
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/dc/tax/income/credits/ptc/fraction_elderly.yaml:27]: image

**Review by MaxGhenis — APPROVED** [review]

**Comment by MaxGhenis** [comment]
Thanks, looks good! And this marks our first fully uprated state income tax model 🎉
